### ISSUE TITLE
Layer:fill with topological problems in the data

### ISSUE THREAD

**Issue by pedro-andrade-inpe** [TerraME/terrame#1568]
`Layer:fill` erases output geometries when the input data has some topological problems. See the data from Gilney and investigate how to solve that.

**Comment by pedro-andrade-inpe** [comment]
@avancinirodrigo, please use the data below to reproduce this problem. Verify if the same problem occurs in TerraView.

[remove_cells.zip](https://github.com/TerraME/terrame/files/777968/remove_cells.zip)

**Comment by avancinirodrigo** [comment]
@pedro-andrade-inpe, for any operation?

**Comment by avancinirodrigo** [comment]
According to Felipe (TerraLib), `coverage` operation does not work when the attribute selected is double type. 
In the TerraView, there is a help with the attribute type for each operation. 
I think that is possible to check the attribute type in TerraLib.

**Comment by avancinirodrigo** [comment]
Yes, this function was removed in save refactoring:
```lua
local function getPropertiesTypes(dse)
	dse:moveFirst()
	local types = {}
	local numProps = dse:getNumProperties()

	for i = 0, numProps - 1 do
		types[dse:getPropertyName(i)] = dse:getPropertyDataType(i)
	end

	return types
end
```

**Comment by pedro-andrade-inpe** [comment]
The solution for this issue will be implemented by #1691.
